src/gcp/cloudscheduler.ts

```diff
@@ -12,38 +14,113 @@ export interface Schedule {
-  const strippedName = schedule.name.replace(/\/[^\/]+$/, "");
+  const strippedName = job.name.replace(/\/[^\/]+$/, "");
   return api.request("POST", `/${VERSION}/${strippedName}`, {
     auth: true,
     origin: api.cloudschedulerOrigin,
-    data: Object.assign({ timeZone: "America/Los_Angeles" }, schedule),
+    data: Object.assign({ timeZone: DEFAULT_TIME_ZONE }, job),
   });
 }
 
+/*
+* Deletes a cloudScheduler job with the given name.
+* Returns a 404 if no job with that name exists.
+* @param name: The name of the job to delete
+*/
 export function deleteJob(name: string): Promise<void> {
   return api.request("DELETE", `/${VERSION}/${name}`, {
     auth: true,
     origin: api.cloudschedulerOrigin,
   });
 }
 
-export async function createOrReplaceJob(schedule: Schedule): Promise<void> {
-  const jobName = `${schedule.name.split("/")[5]}`;
+/*
+* Gets a cloudScheduler job with the given name.
+* If no job with that name exists, this will return a 404.
+* @param name: The name of the job to get
+*/
+export function getJob(name: string): Promise<any> {
+  return api.request("GET", `/${VERSION}/${name}`, {
+    auth: true,
+    origin: api.cloudschedulerOrigin,
+  });
+}
+
+/*
+* Updates a cloudScheduler job.
+* Returns a 404 if no job with that name exists.
+* @param job: A job to update. Note that name cannot be updated.
+*/
+export function updateJob(job: Job): Promise<any> {
+  return api.request("PATCH", `/${VERSION}/${job.name}`, {
+    auth: true,
+    origin: api.cloudschedulerOrigin,
+    data: Object.assign({ timeZone: DEFAULT_TIME_ZONE }, job),
+  });
+}
+
+/*
+* Checks for a existing job with the given name.
+* If none is found, it creates a new job.
+* If one is found, and it is identical to the job parameter, it does nothing.
+* Otherwise, if one is found and it is different from the job param, it updates the job.
+* @param job: A job to check for and create, replace, or leave as appropriate
+*/
+export async function createOrReplaceJob(job: Job): Promise<any> {
+  const jobName = `${job.name.split("/")[5]}`;
   try {
-    await createJob(schedule);
-    logLabeledSuccess("functions", `created scheduler job ${jobName}`);
+    const existingJob = await getJob(job.name);
+    if (!job.timeZone) {
+      // We set this here to avoid recreating schedules that use the default timeZone
+      job.timeZone = DEFAULT_TIME_ZONE;
+    }
+
+    if (isIdentical(existingJob.body, job)) {
+      logLabeledBullet("functions", `scheduler job ${jobName} is up to date, no changes required`);
+      return;
+    } else {
+      logLabeledBullet("functions", `updating scheduler job ${jobName}`);
+      return updateJob(job);
+    }
   } catch (e) {
-    if (e.context.response.statusCode !== 409) {
+    // if the error status is 404, no job exists, so we can create one
+    // if it is anything else, we should error out
+    if (e && e.context && e.context.response && e.context.response.statusCode !== 404) {
       throw e;
```

Comment:
I'm assuming this will get returned to the user as an error when it throws? If so, we should wrap this in a `FirebaseError` with a friendly message.

Reply:
The CLI error handling will automatically surface error messages coming from Google APIs, so you hopefully shouldn't need to do anything special here.